README.md

```diff
@@ -1,15 +1,52 @@
-# Hapi
+# HAPI Protocol
 
-This is a crate that helps implement the use of the [HAPI protocol](https://hapi-one.gitbook.io/hapi-protocol/) in your smart contract on Near blockchain.
+[HAPI] is a one-of-a-kind decentralized security protocol that prevents and interrupts any potential malicious activity within the blockchain space. HAPI works by leveraging both external and off-chain data as well as on-chain data accrued directly by HAPI and is publicly available.
 
-The crate has an AML structure that stores:
+# HAPI NEAR Proxy
 
-- account_id: AccountId - the address of the [aml contract](https://github.com/HAPIprotocol/near-proxy-contract);
+It is a proxy [contract](https://github.com/HAPIprotocol/near-proxy-contract) used for reporting and receiving data from [HAPI] protocol main contract on the NEAR blockchain.
+
+Address of the contract on the **testnet**
+
+```
+contract.hapi-test.testnet
+```
+Address of the contract on the **mainnet**
+
+```
+proxy.hapiprotocol.near
+```
+
+# HAPI NEAR Connector
+
+This crate helps implement the use of the [HAPI] in your smart contract on the NEAR blockchain. You can use it to make request data from the proxy contract, and subsequent processing this data.
```

Comment:
"This crate helps to implement HAPI Protocol in your smart contract..." would sound a bit cleaner.